File: app/src/main/java/com/alexanderageychenko/ecometer/Logic/dagger2/AppComponent.java
package com.alexanderageychenko.ecometer.Logic.dagger2;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by Alexander on 15.04.2017.
 */
@Singleton
@Component(modules={DepositoryModule.class, AppModule.class})
public interface AppComponent {
    Getter getGetter();
    Injector getInjector();
}

File: app/src/main/java/com/wackalooon/ecometer/di/AppComponent.kt
package com.wackalooon.ecometer.di

import com.wackalooon.ecometer.EcoMeterApp
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        ActivityModule::class,
        AppModule::class,
        DaoModule::class,
        MapperModule::class,
        NetworkModule::class,
        RepositoryModule::class,
        UseCaseModule::class,
        ViewModelModule::class,
        AndroidSupportInjectionModule::class
    ]
)
interface AppComponent : AndroidInjector<EcoMeterApp> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<EcoMeterApp>()
}
